bundler/src/lib.rs

```diff
@@ -82,9 +81,8 @@ impl Bundler {
     async fn write_car(self) -> Result<()> {
         let mut out = async_std::fs::File::create(&self.bundle_dst).await?;
 
-        // Invert the actor index so that it's CID => Type.
-        let manifest: Manifest = self.added.into_iter().map(|(typ, cid)| (cid, typ)).collect();
-
+        let manifest: Vec<(String, Cid)> =
+            self.added.iter().map(|(t, c)| (String::from(t), c.clone())).collect();
```

Comment:
```suggestion
            self.added.iter().map(|(t, c)| (t.into(), *c)).collect();
```

Reply:
The fact that the ordering is _implicitly_ based on the actor "types" is a bit strange. We should either:

1. Call that out.
2. Sort by name (lexicographically).

Reply:
well, not all that strange. We can call it out, but it really is non-normative.
Or we can just say the order is arbitrary.

Reply:
Also, what is this into business? I can understand dereferencing instead of cloning, but that into is downright strange.

Reply:
I'll keep the String::from so that it is immediately evident by the code what is happening, but use the dereference.

Reply:
`into()` is a member of the `Into` trait. It performs the conversion using Rust's type inference system.

There's a blanket implementation of `Into` for types that implement `From`: https://doc.rust-lang.org/std/convert/trait.Into.html#impl-Into%3CU%3E. That's the magic here.

Once you train your Rust eye, it is very simple to read because the `String` type is already present in the type `Vec<(String, Cid)>`.

Reply:
I don't have strong feelings here, but concur with @Stebalien that the `into()` version feels more idiomatic. We should aim to make code more Rust-idiomatic than "let's satisfy Vyzo's liking today" (a liking which will evolve as you get more used to Rust).

Reply:
ok, ok, the litte crab has been battered to submission by some giant claws.
into it is then.